src/main/java/htmlpublisher/util/MultithreadedDirScanner.java

```diff
@@ -0,0 +1,51 @@
+package htmlpublisher.util;
+
+import hudson.util.DirScanner;
+import hudson.util.FileVisitor;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import java.io.File;
+import java.io.IOException;
+
+public class MultithreadedDirScanner extends DirScanner {
+
+	private DirScanner delegateDirScanner;
+	private int numberOfThreads;
+
+	public MultithreadedDirScanner(DirScanner delegateDirScanner, int numberOfThreads) {
+		this.delegateDirScanner = delegateDirScanner;
+		this.numberOfThreads = numberOfThreads;
+	}
+
+	@Override
+	public void scan(File dir, FileVisitor visitor) throws IOException {
+		// Create new executor service
+		ExecutorService executorService = Executors.newFixedThreadPool(this.numberOfThreads);

```

Comment:
would be easier to use the jenkins core utility for this see class `jenkins.util.Timer` and so avoid managing your own ExecutorService

Reply:
When using `jenkins.util.Timer.get()`, I get the existing Jenkins-ExecutorService.
How can I set the _exact number of threads_ for executing the scan tasks for that already instanced ExecutorService?

Reply:
The code has been refactored: 94bbb3aa44aff139de9b9f6db52e39834564ba83
* Introduced `htmlpublisher.util.FileVisitorService`:
  * manages a queue, that collects the scanned files
  * uses a worker pool for visiting the scanned files:
    * the worker pool is managed using the Jenkins ExecutorService (`jenkins.util.Timer.get()`)
    * collected files will be delegated to original `FileVisitor.visit()` by one worker of the pool
  * handles (final) shutdown after the end of scan and exceptions during visits

Reply:
@olamy
Please review the changes. Thx!